lang/cs/Org.Apache.REEF.Client/Common/HttpClientHelper.cs

```diff
@@ -90,12 +108,17 @@ internal static List<string> GetRmUri(string filePath)
             }
         }
 
-        internal static string GetAppId(string filePath)
+        /// <summary>
+        /// Get application id for the Job submitted 
+        /// </summary>
+        /// <param name="filePath"></param>
+        /// <returns></returns>
+        internal string GetAppId(string filePath)
         {
             using (var sr = new StreamReader(File.Open(filePath, FileMode.Open, FileAccess.Read, FileShare.Read)))
             {
-                var appId = sr.ReadLine();                
-                return appId;
+                _appId = sr.ReadLine();
```

Comment:
appId was written to the file in REEF Java code after the Job is submitted to RM. It is unique per Job Submit. I can change the code to read it only once instead of reading every time it is called.